api/hasura/cron/epochs.ts

```diff
@@ -725,6 +813,92 @@ async function notifyEpochStatus(
   return true;
 }
 
+function calculateNumReceived(
+  epoch: Pick<EpochsToNotify, 'endEpoch'>['endEpoch'][number]
+) {
+  return (epoch.circle?.users || [])
```

Comment:
this could also be another reduce, iterate over this collection once, if there is an email, push the results of `tokenNumReceived` and `notesNumReceived` to the accumulator. @sshmm